config/v1/types_infrastructure.go

```diff
@@ -620,9 +620,36 @@ const (
 	AzureStackCloud AzureCloudEnvironment = "AzureStackCloud"
 )
 
+// GCPServiceEndpoint store the configuration of a custom url to
+// override existing defaults of GCP Services.
+type GCPServiceEndpoint struct {
+	// name is the name of the GCP service.
+	// The list of all the service names can be found at https://developers.google.com/apis-explorer
+	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Pattern=`^[a-z0-9-]+$`
+	// +required
+	Name string `json:"name"`
+
+	// url is fully qualified URI with scheme https, that overrides the default generated
+	// endpoint for a client.
+	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Pattern=`^https://`
```

Comment:
As mentioned in my comment on the name field, we now recommend that pattern validations use CEL validations. Same things to consider apply here. Kube has some nice helpers for working with URLs in CEL that may be of interest here: https://kubernetes.io/docs/reference/using-api/cel/#kubernetes-url-library

You mention in the description that this must be a fully qualified URI but I only see a validation for the scheme matching `https://`. Are there further validations you can put in place to ensure that the value matches the expectations of a fully qualified URI? As-is, something like `https://67788272762673737hajsdfha)+)*&D^YF&A*S^F&(^(*&` would be allowed as a value but is clearly not a valid fully qualified URI.